catalog/transformer/catalog-transformer-pdf/src/main/java/ddf/catalog/transformer/input/pdf/PdfInputTransformer.java

```diff
@@ -57,6 +77,28 @@
     private static final FastDateFormat DATE_FORMAT = FastDateFormat.getInstance(
             "yyyy-MM-dd'T'HH:mm:ssZZ");
 
+    private Map<ServiceReference, ContentMetadataExtractor> contentMetadataExtractors =
+            Collections.synchronizedMap(new TreeMap<>(new ServiceComparator()));
+
+    public void addContentMetadataExtractors(
+            ServiceReference<ContentMetadataExtractor> contentMetadataExtractorRef) {
+        Bundle bundle = getBundle();
```

Comment:
It's not obvious to me why we're going to the `BundleContext` with all these `ServiceReference`s instead of just injecting the services normally via blueprint. The `ServiceReference` key only seems to be used for updating the map. Might be worth a comment, since this pattern is normally discouraged.